### PR TITLE
fix `jsx-prefer-fragment-wrappers`

### LINKED ISSUE #111: `jsx-prefer-fragment-wrappers` returns invalid for lots of HTML markup
Consider the following example:
```
function Table({content, otherContent}) {
 return (
   <table>
     <tbody>
       <tr>{content}</tr>
       <td>{otherContent}</td>
     </tbody>
   </table>
  )
}
```

The rule currently would consider all of those tags invalid. I think we should remove all the tag checking logic to just look for `divs` and `spans`.

@lemonmade what do you think?

### PR DESCRIPTION
**Pull request by cartogram**
closes https://github.com/Shopify/eslint-plugin-shopify/issues/111

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ This plugin provides the following custom rules, which are included as appropria
 - [jquery-dollar-sign-reference](docs/rules/jquery-dollar-sign-reference.md): Require that all jQuery objects are assigned to references prefixed with `$`.
 - [jsx-no-complex-expressions](docs/rules/jsx-no-complex-expressions.md): Disallow complex expressions embedded in in JSX.
 - [jsx-no-hardcoded-content](docs/rules/jsx-no-hardcoded-content.md): Disallow hardcoded content in JSX.
-- [jsx-prefer-fragment-wrappers](docs/rules/jsx-no-hardcoded-content.md): Disallow useless wrapping elements in favour of fragment shorthand in JSX.
+- [jsx-prefer-fragment-wrappers](docs/rules/jsx-prefer-fragment-wrappers.md): Disallow useless wrapping elements in favour of fragment shorthand in JSX.
```
Comment (cartogram) [inline README.md:122]: unrelated typo

**Review by BPScott — APPROVED** [review]
LGTM.

Do we need a changelog entry?

File: lib/rules/jsx-prefer-fragment-wrappers.js [hunk 2]
```diff
@@ -35,23 +33,15 @@ module.exports = {
   },
 };
 
-function isPascalCase(string) {
-  return string === pascalCase(string);
+function isAcceptedTagName({openingElement: {name}}) {
+  return ['div', 'span'].every((tagName) => tagName !== name.name);
```
Comment (lemonmade) [inline lib/rules/jsx-prefer-fragment-wrappers.js]: IMO `![].some` is probably more readable here, or inverting the whole function's purpose, it is hard for me to read "accepted tag name, but we are checking that ever non-accepted tag name is not the case"

**Review by lemonmade — APPROVED** [review]

File: lib/rules/jsx-prefer-fragment-wrappers.js (hunk 2 above)
Comment (cartogram) [reply lib/rules/jsx-prefer-fragment-wrappers.js]: I like `![].some` and keeping this purpose consistent with the others like it in the file.

**Pull request merged**